### ISSUE TITLE
Versioning accept header strict/cascade settings appears to not work in this case

### ISSUE THREAD

**Issue by elliotlarson** [ruby-grape/grape#1082]
If I have this config for versioning...

```ruby
version 'v1', using: :header, vendor: 'acme', strict: true, cascade: false
```

then this should require that I pass the appropriate `Accept` request header to get a valid `200` response.

This request should work:

    $ curl -H 'Accept:application/vnd.acme-v1+json' http://localhost:3000/api/projects
    => succeeds √ 

... and it does, returning a successful `200` response.

But, not passing the correct Accept header should result in an error.  For example, this request:

    $ curl -H 'Accept:application/vnd.acme-nonexistentversion+json' http://localhost:3000/api/projects
    => fails √ 

... fails with a `406` response as expected.

However, this request returns a successful `200` API response, which is unexpected:

    $ curl -H 'Accept:application/xml' http://localhost:3000/api/projects
    => succeeds ?

I didn't make the request with the correct version `Accept` header, but I'm getting a successful `200` response.  Is this a bug, or have I misunderstood how the `strict: true, cascade: false` config settings are supposed to work?

**Comment by dblock** [comment]
I believe your logic is correct. Without a valid `Accept` header you have no matching version and it should fail with a 406 as long as `strict: true`. Lets get to the bottom of this, try writing a test that reproduces it on Grape's HEAD?

**Comment by elliotlarson** [comment]
I'm a little confused by the way the headers are being setup in the relevant tests.  In the spec file `spec/grape/exceptions/invalid_accept_header_spec.rb`, all of the specs that pass in an `Accept` header to rack-test, do it like this:

```ruby
get '/beer', {}, 'HTTP_ACCEPT' => 'application/vnd.invalidvendor-v99' 
```

It's not clear to me why the "HTTP_ACCEPT" header is used instead of "Accept".  Shouldn't these tests be written like so:

```ruby
get '/beer', {}, 'Accept' => 'application/vnd.invalidvendor-v99' 
```

**Comment by elliotlarson** [comment]
Okay, I created a pull request for this: https://github.com/ruby-grape/grape/pull/1101  At least, this seemed fairly straight forward to test and fix.  It looks like if the accept header was a media type, it was failing to catch the missing version and return a 406.

**Comment by dm1try** [comment]
Fixed in #1101.
